docs/configuration_files.rst

```diff
@@ -0,0 +1,350 @@
+
+      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
+        z: !numpy.linspace {start: 0, stop: 1.3, num: 10}
+
+  `N.B.` To call a function with no arguments, you should pass an empty list of
+  ``args`` or an empty dictionary of ``kwargs``. For example:
+
+  .. code:: yaml
+
+      cosmo: !astropy.cosmology.default_cosmology.get []
+
+
+* `Reference a variable`: variables can be referenced by their full name tagged with a dollar sign ``$``.
+  In the previous example you could also define the variables at the top-level of the file and then reference them:
+
+  .. code:: yaml
+
+      redshift: !numpy.linspace [ 0, 1.3, 10 ]
+      comoving_distance: !astropy.cosmology.Planck15.comoving_distance
+        z: $redshift
+
+  Please, check below in the `cosmology`_ section how to use different cosmologies with the SkyPy pipeline.
+
+
+Tables
+^^^^^^
+
+* `Create a table`: a dictionary of table names, each resolving to a dictionary of column names for that table.
+
+  Let us create a table called ``telescope`` with a column to store the width of spectral lines that follow a normal distribution
+
+  .. code:: yaml
+
+      tables:
+        telescope:
+          spectral_lines: !scipy.stats.norm
+            loc: 550
+            scale: 1.6
+            size: 100
+
+* `Add a column`: you can add as many columns to a table as you need.
+    Imagine you want to add a column for the telescope collecting surface
+
+  .. code:: yaml
+
+      tables:
+        telescope:
+          spectral_lines: !scipy.stats.norm
+            loc: 550
+            scale: 1.6
+            size: 100
+          collecting_surface: !numpy.random.uniform
+            low: 6.9
+            high: 7.1
+
+* `Reference a column`: columns in the pipeline can be referenced by their full name tagged with a dollar sign ``$``.
+  Example: the radius of cosmic voids seem to follow a lognormal distribution. You can create a table ``cosmic_voids``
+  with a column ``radii`` where you sample 10000 void sizes and two other columns, ``mean`` and ``variance`` that reference
+  the first column
+
+  .. code:: yaml
+
+    tables:
+      cosmic_voids:
+        radii: !scipy.stats.lognorm.rvs
+          s: 200.
+          size: 10000
+        mean: !numpy.mean
+          a: $cosmic_voids.radii
+        variance: !numpy.var
+          a: $cosmic_voids.radii
+
+
+* `Multi-column assignment`: if a function returns multiple columns, you can chose to assign them to multiple columns with different names or to a muti-column object.
+
+  Imagine you want the distribution of the circular velocities of 1000 halos following a Maxwellian distribution.
+  The histogram NumPy_ returns a 2-dimensional object. Therefore, you can choose
+
+  .. code:: yaml
+
+    tables:
+      halos:
+        circular_velocities: !scipy.stats.maxwell.rvs
+          s: 250
+          size: 1000
+        hist, bin_edges: !numpy.histogram
+          a: $halos.circular_velocities
+          density: True
```

Comment:
- `s` is not a valid keyword argument for `scipy.stats.maxwell.rvs`
- `hist` and `bin_edges have different lengths from each other and from `circular_velocities` and so cannot be valid columns in the same table
- Conceptually this example makes no sense. It is supposed to be a table where each row represents a dark matter halo and each column is a property of each halo e.g. mass, circular velocity, velocity dispersion, number of subhalos etc. A histogram (and the bin edges) is not a halo property!